src/app/modules/main/chat_section/module.nim

```diff
@@ -279,8 +277,6 @@ proc initContactRequestsModel(self: Module) =
   self.view.contactRequestsModel().addItems(contactsWhoAddedMe)
 
 proc rebuildCommunityTokenPermissionsModel(self: Module) =
-  self.buildTokenList()
```

Comment:
Keep in mind that `rebuildCommunityTokenPermissionsModel` is also triggered when the collectibles or wallet state changes.

Where do we still ensure that the token lists is rebuilt on those events?

Reply:
Please double-check if it will not break Assets list:
1. Create the community
2. Add permission
3. Add new permission
4. Assets are present

Reply:
Tested the flow above and it works

Reply:
> we need to re-check if the permissions are met, but it's not necessary to rebuild the permission model itself when we get token updates.

This is correct. The important part is that:

1. That the token/collectible model is built **before** the token permission model
2. The token/collectible model is rebuilt/updated when there's an update in the wallet (which then also re-triggers permission checks and ultimately rebuilds/updates the permission model

Reply:
> That the token/collectible model is built before the token permission model

Is that really a requirement? I don't think they have a dependency on each other.

The permission checks are made in status-go or at least in the service, so the model doesn't matter.

Let me know if I'm missing something.

If you're right, then I need to fix my PR because there order is not guaranteed with my changes.

Reply:
> Is that really a requirement? I don't think they have a dependency on each other.

Unfortunately they do. The view components have been built in a way that the permissions can only render properly when an additional token model was supplied.

But they do update when the model updates, so as soon as the token model is ready, it should render the permissions fine as well. But until that point you might see broken token permissions (which might be a fraction of a second depending on how the planets align)

Reply:
Sure :D